python/lsst/daf/butler/registry/queries/exprParser/exprTree.py

```diff
@@ -317,6 +317,32 @@ def __str__(self) -> str:
         return "({expr})".format(**vars(self))
 
 
+class TupleNode(Node):
+    """Node representing a tuple, sequence of parenthesized expressions.
+
+    Tuple is used to represent time ranges, for now parser supports tuples
+    with two items, though this class can be used to represent different
+    number of items in sequence.
+
+    Attributes
+    ----------
+    items : tuple of Node
+        Expressions inside parentheses.
+    """
+    def __init__(self, items: Tuple[Node]):
```

Comment:
I think the type annotation here should either be `Tuple[Node, ...]` (for any number of elements) or `Tuple[Node, Node]` (for 2-tuples only).  I think you're supporting only 2-tuples in the syntax here, but I wasn't sure if you were trying to make the visitor interface extensible to more later.

Reply:
I'll add `...`, indeed the language grammar now allows only two items, but parsed tree is separate from grammar so you can build your own expression trees, and I can't exclude we can extend grammar with longer tuples if needed.